tests/conftest.py

```diff
@@ -47,12 +34,40 @@ def tear_down():
             conn.execute('rollback')
             if 'mssql' in db_url:
                 conn.connection.connection.autocommit = True
-            conn.execute('create database %s' % TEST_DB)
+            conn.execute('create database %s' % db_name)
     else:
-        raise Exception('Database %s already exists; refusing to overwrite' % TEST_DB)
+        raise Exception('Database %s already exists; refusing to overwrite' % db_name)
 
     request.addfinalizer(tear_down)
 
     params = request.param.copy()
     params['url'] = db_connection_url
     return params
+
+
+@pytest.fixture(scope="class", params=[
+    {
+        'url': "postgresql://postgres@localhost/%s",
+        'admin_db': 'postgres'
+    },
+    {
+        'url': 'mysql+pymysql://travis@/%s?charset=utf8',
+    },
+    {
+        'url': 'mssql+pyodbc://SA:Password@123@localhost/%s?driver=ODBC+Driver+17+for+SQL+Server',
+        'admin_db': 'master'
+    }
+], ids=['postgres', 'mysql', 'mssql'])
```

Comment:
I'm new to `pytest`, would you mind explaining a bit about how this works?  From the docs it sounds like these `@pytest.fixture`s use dependecy injection to set up test data.  What is this `request` object - some built-in thing? Do the params specified here get set on that object?

Reply:
yes that's right. This will create the fixture 3 times, one for each set of params and any test that uses this fixture will get run once for each fixture.

The injection is based on the function name matching the test arg name.

Reply:
Cool, thanks!